ci-operator/step-registry/workers-rhel/azure-provision/workers-rhel-azure-provision-commands.sh

```diff
@@ -62,6 +62,9 @@ for count in $(seq 1 ${RHEL_WORKER_COUNT}); do
     echo "Unable to get ip of rhel instance ${infra_id}-rhel-${count}!"
     exit 1
   fi
+   
+  #Missing identity prevents VM from accessing storage accounts in Azure API
```

Comment:
Maybe we can directly add `--assign-identity` option to skill this command ?

Reply:
ok, seems `--assign-identity` in `az vm create ` is expecting the resource id instead of name, let me find a cluster to test the output of `az vm identity show `

> --assign-identity
> Accept system or user assigned identities separated by spaces. Use '[system]' to refer system assigned identity, or a resource id to refer user assigned identity. Check out help for more examples.
>

Reply:
I guess `--assign-identity` option from `az vm create` and `--identities` option from `az vm identity assign` should be able to receive the same value, you can test locally (pass id name to  `--assign-identity` ) to see if it works. If not, only expect id, then I think we do not need to introduce a dedicated command to retrieve its id, we and keep the present `az vm identity assign` command line.

Reply:
yes, the test result looks indeed so 👍 
With `gpei-1107-7shkz-identity` set in `--assign-identity`, `userAssignedIdentities` created as expected. 

```
# az vm create --resource-group gpei-1107-7shkz-rg --name gpei-test-rhel-0 --image ...
...
 --assign-identity gpei-1107-7shkz-identity
```

```
# az vm identity show --name gpei-test-rhel-0  --resource-group gpei-1107-7shkz-rg
{
  "principalId": null,
  "tenantId": null,
  "type": "UserAssigned",
  "userAssignedIdentities": {
    "/subscriptions/xxxxxxx/resourceGroups/gpei-1107-7shkz-rg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/gpei-1107-7shkz-identity": {
      "clientId": "xxxxx",
      "principalId": "xxxxx"
    }
  }
}
```